htmd/molecule/molecule.py

```diff
@@ -2040,6 +2059,55 @@ def test_uniqueResidueID(self):
         assert r1 == r2
         assert r2 != r3
 
+    def test_selfalign(self):
+        from htmd.home import home
+
+        # Checking bonds
+        mol = self.trajmol.copy()
+        _ = mol.filter('resname MOL')
+
+        mol.align('noh')
+
+        refcoords = np.load(path.join(home(dataDir='test-molecule'), 'test-selfalign-mol.npy'))
+
+        assert np.allclose(mol.coords, refcoords, atol=1E-6)
+
+    def test_alignToReference(self):
+        from htmd.home import home
+
+        # Checking bonds
+        mol = self.trajmol.copy()
+        _ = mol.filter('resname MOL')
+
+        mol2 = mol.copy()
+        mol2.dropFrames(keep=3)  # Keep a random frame
+        _ = mol2.filter('noh') # Remove some atoms to check aligning molecules with different numAtoms
+
+        mol.align('noh', refmol=mol2)
+
+        refcoords = np.load(path.join(home(dataDir='test-molecule'), 'test-align-refmol.npy'))
+
+        assert np.allclose(mol.coords, refcoords, atol=1E-6)
+        assert np.allclose(mol.coords[mol.atomselect('noh'), :, 3], mol2.coords[:, :, 0], atol=1E-6)
+
+    def test_alignToReferenceMatchingFrames(self):
```

Comment:
The `matchingframes` is a new argument, I thought we should be testing `frames` as well

Reply:
Done